src/extension.ts

```diff
@@ -69,6 +77,25 @@ export function activate (context: vscode.ExtensionContext) {
   commandManager.register(new commands.SaveHTML(engine))
   commandManager.register(new commands.SaveDocbook(engine))
 
+  const antora = new AntoraSupport()
+  if (await antora.isDetected()) {
+    const workspaceConfiguration = vscode.workspace.getConfiguration('asciidoc', null)
+    // look for Antora settings in long term storage
+    const workspaceState: vscode.Memento = context.workspaceState
+    const isEnableAntoraSupportSettingDefined = await workspaceState.get('antoraSupportSetting')
+    if (workspaceConfiguration.get('antora.enableAntoraSupport') === false && isEnableAntoraSupportSettingDefined === undefined) {
+      const answer = await vscode.window.showInformationMessage('We detect that you are working with Antora. Do you want to active Antora support?', 'yes', 'no thanks')
+      if (answer !== undefined) {
+        await workspaceState.update('antoraSupportSetting', true)
+        const enableAntoraSupport = answer === 'yes' ? true : (answer === 'no thanks' ? false : undefined)
+        await workspaceConfiguration.update('antora.enableAntoraSupport', enableAntoraSupport)
+        if (enableAntoraSupport) {
+          context.subscriptions.push(vscode.languages.registerCompletionItemProvider(selector, new AntoraCompletionProvider(), '{'))
+        }
+      }
+    }
+  }
```

Comment:
We should move this logic when an AsciiDoc file is opened using the listener `onDidOpenTextDocument` on a `workspace`.
As a result, we can ask to enable Antora support only when we find a valid `antora.yml` for the given `TextDocument` 🎉